platform/security/sts/security-sts-ldaplogin/src/main/java/ddf/ldap/ldaplogin/LdapLoginConfig.java

```diff
@@ -79,12 +111,128 @@
+            createLdapConnectionFactory(
+                (String) ldapProperties.get(LDAP_URL),
+                Boolean.parseBoolean((String) ldapProperties.get(START_TLS)));
+
+        ldapConnectionPool =
+            new GenericObjectPool<>(
+                new LdapConnectionPooledObjectFactory(ldapConnectionFactory),
+                createGenericPoolConfig(id),
+                createGenericPoolAbandonConfig());
+
+      } catch (LdapException e) {
+        LOGGER.error("Error creating ldap connection factory", e);
+      }
+
+      // TODO Switch to DictionaryMap.java
+      Dictionary<String, String> serviceProps = new Hashtable<>();
+      serviceProps.put("id", id);
+
+      LOGGER.debug("Registering LdapConnectionPool");
+      connectionPoolServiceRegistration =
+          context.registerService(
+              GenericObjectPool.class.getName(), ldapConnectionPool, serviceProps);
       // create modules from the newly updated config
       Module ldapModule = createLdapModule(props);
       ldapService.update(ldapModule);
     }
   }
 
+  private AbandonedConfig createGenericPoolAbandonConfig() {
+    AbandonedConfig abandonedConfig = new AbandonedConfig();
+    abandonedConfig.setRemoveAbandonedOnBorrow(true);
+    abandonedConfig.setRemoveAbandonedTimeout(FIVE_MIN_S);
+    return abandonedConfig;
+  }
+
+  private GenericObjectPoolConfig createGenericPoolConfig(String id) {
+    GenericObjectPoolConfig config = new GenericObjectPoolConfig();
+    config.setJmxNameBase("org.apache.commons.pool2:type=LDAPConnectionPool,name=");
+    config.setJmxNamePrefix(id);
+    config.setTimeBetweenEvictionRunsMillis(FIVE_MIN_MS);
+    config.setTestWhileIdle(true);
+    config.setTestOnBorrow(true);
+    return config;
+  }
+
+  protected LDAPConnectionFactory createLdapConnectionFactory(String url, Boolean startTls)
+      throws LdapException {
+    boolean useSsl = url.startsWith("ldaps");
+    boolean useTls = !url.startsWith("ldaps") && startTls;
+
+    Options lo = Options.defaultOptions();
+
+    try {
+      if (useSsl || useTls) {
+        LOGGER.trace("Setting up secure LDAP connection.");
+        initializeSslContext();
+        lo.set(LDAPConnectionFactory.SSL_CONTEXT, getSslContext());
+      } else {
+        LOGGER.trace("Setting up insecure LDAP connection.");
+      }
+    } catch (GeneralSecurityException e) {
+      LOGGER.info("Error encountered while configuring SSL. Secure connection will fail.", e);
+    }
+
+    lo.set(LDAPConnectionFactory.HEARTBEAT_TIMEOUT, new Duration(30L, TimeUnit.SECONDS));
+    lo.set(LDAPConnectionFactory.HEARTBEAT_INTERVAL, new Duration(60L, TimeUnit.SECONDS));
+    lo.set(LDAPConnectionFactory.CONNECT_TIMEOUT, new Duration(30L, TimeUnit.SECONDS));
+
+    lo.set(LDAPConnectionFactory.SSL_USE_STARTTLS, useTls);
+    lo.set(
+        LDAPConnectionFactory.SSL_ENABLED_CIPHER_SUITES,
+        Arrays.asList(System.getProperty("https.cipherSuites", "").split(",")));
```

Comment:
I believe this is re-used code rather than "new" code, but it is probably a good idea to check that these properties exist before using them in the very unlikely case they aren't set to avoid an NPE.  Also you can use the constants already defined in https://github.com/codice/ddf/blob/master/platform/security/platform-security-core-api/src/main/java/ddf/security/SecurityConstants.java